spec/unit/blockscore/question_set_spec.rb

```diff
@@ -1,62 +1,210 @@
+
+        let(:qs) { QuestionSet.new({person_id: person.id, id: resource_id}) }
+        let(:data) { person.question_sets.data }
+        subject { person.question_sets.retrieve(qs.id) }
+        before(:each) do
+          person.question_sets.data.clear
+          allow(QuestionSet).to receive(:retrieve).with(qs.id) { qs }
+        end
+
+
+        it "question_sets uses QuestionSet.retrieve" do
+          expect(QuestionSet).to receive(:retrieve).with(qs.id) { qs }
+          expect(subject).to be(qs)
+        end
+
+        it "registers new question set" do
+          aggregate_failures("for person and collection") do
+            expect(qs).to receive(:id).at_least(:once).and_call_original
+            expect(person.question_sets).to receive(:<<).with(qs).and_call_original
+            expect(data).to receive(:<<).with(qs.id).and_call_original
+          end
+          person.question_sets.retrieve(qs.id)
+        end
+
+        it "errors if not belonging" do
+          qs.person_id = "some_not_associated_id"
+          expect { subject }.to raise_error(Error, "None belonging")
+        end
+
+      end
     end
 
-    it 'retrieve' do
-      qs = person.question_sets.create
-      person.question_sets.retrieve(qs.id)
-      assert_requested(api_stub, times: 3)
+
+    context "when id is invalid" do
+      pending "should raise not found if not a resource style id" do
+        expect { person.question_sets.retrieve("bad_id") }.to raise_error(NotFoundError)
+      end
+
+      it "should raise ArgumentError if empty" do
+        expect { person.question_sets.retrieve("") }.to raise_error(ArgumentError)
+      end
     end
 
-    describe '#all' do
-      it 'requests' do
-        person.question_sets.all
-        assert_requested(api_stub, times: 2)
+
+    describe "#new" do
+
+      subject { person.question_sets }
+
+      it "should delegate to QuestionSet" do
+        expect(QuestionSet).to receive(:new).with(subject.default_params).and_call_original
+        result = person.question_sets.new
+        expect(result).to be_an_instance_of(QuestionSet)
       end
 
-      it ':count' do
-        response = person.question_sets.all(count: 2)
-        expect(response.count).to eq(2)
-        assert_requested(api_stub, times: 2)
+      it "should save person if not save" do
+        person = Person.new
+        qs = person.question_sets.new
+        expect(person).to receive(:save).and_call_original
+        qs.save
       end
 
-      it ':offset' do
-        response = person.question_sets.all(count: 2, offset: 2)
-        expect(response.count).to eq(2)
-        assert_requested(api_stub, times: 2)
+      it "should have person#id" do
+        qs = person.question_sets.new
+        qs.person_id = "some_id"
+        qs.save
+        expect(qs.person_id).to eq(person.id)
+        expect(qs.id).not_to be(nil)
       end
+
+      it "should add to question_sets" do
+        count = subject.size
+        qs = subject.new
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.